File: 1033 Moving Stones Until Consecutive.py
class Solution:
    def numMovesStones(self, a: int, b: int, c: int) -> List[int]:
        if a > b:
            a, b = b, a
        if c < a:
            a, b, c = c, a, b
        elif c < b:
            b, c = c, b
        
        res = [0, 0]
        if a == b - 1 and b == c - 1:
            return res
        if a == b - 1 or b == c - 1:
            res[0] = 1
        elif a == b - 2 or b == c - 2:
            res[0] = 1
        else:
            res[0] = 2
        
        res[1] = c - a - 2
        
        return res

File: 1033 Moving Stones Until Consecutive.java
class Solution {
    public int[] numMovesStones(int a, int b, int c) {
        int z = Math.max(a, Math.max(b, c));
        int x = Math.min(a, Math.min(b, c));
        int y = a + b + c - x - z;
        int[] res = new int[2];
        
        if (z - y == 1 && y - x == 1) {
            return res;
        } else if (z - y == 1 || y - x == 1) {
            res[0] = 1;
        } else if (z - y == 2 || y - x == 2) {
            res[0] = 1;
        } else {
            res[0] = 2;
        }
        
        res[1] = z - x - 2;
        return res;
    }
}